### DISCUSSION TITLE
Terminal Options

### DISCUSSION THREAD

**Discussion by bombardier200** [migueldeicaza/SwiftTerm#278] [Q&A]
If I have a swift application that uses SwiftTerm and specifically LocalProcessTerminalView, how can I edit TerminalOptions?

**Comment by migueldeicaza** [comment]
I have not added an API to update this value dynamically.

I am not sure if it is better to have a single API to update everything at once, or instead add individual APIs for every property you might want to change dynamically.   I think the latter is better.

**Comment by bombardier200** [comment]
Yeah I might take a look at this as for CodeEdit we want to be able to change things like scrollBack. I noticed that you implement calls for things like the name and for cursor style to update them, so, maybe just adding functions for the other features in TerminalOptions would suffice?

**Reply by migueldeicaza** [reply]
I would love to implement the scrollback change.   I think it is best done as individual properties as opposed to passing all the options in one go.  

L

**Reply by bombardier200** [reply]
Yeah I agree, I am on vacation till next week but when I get back Ill take a look!
